website/addons/dataverse/client.py

```diff
@@ -1,89 +1,95 @@
+from dataverse.exceptions import ConnectionError, UnauthorizedError, OperationFailedError
 
 from framework.exceptions import HTTPError
 from website.addons.dataverse import settings
 
 
-def connect(username, password, host=settings.HOST):
-    connection = Connection(
-        username=username,
-        password=password,
-        host=host,
-        disable_ssl=not settings.VERIFY_SSL,
-    )
-    return connection if connection.connected else None
+def _connect(token, host=settings.HOST):
+    try:
+        return Connection(host, token)
+    except ConnectionError:
+        return None
 
 
 def connect_from_settings(user_settings):
-    return connect(
-        user_settings.dataverse_username,
-        user_settings.dataverse_password
-    ) if user_settings else None
+    try:
+        return _connect(user_settings.api_token) if user_settings else None
+    except UnauthorizedError:
+        return None
 
 
-def connect_or_403(username, password, host=settings.HOST):
-    connection = Connection(
-        username=username,
-        password=password,
-        host=host,
-        disable_ssl=not settings.VERIFY_SSL,
-    )
-    if connection.status == http.FORBIDDEN:
-        raise HTTPError(http.FORBIDDEN)
-    return connection if connection.connected else None
+def connect_or_401(token):
+    try:
+        return _connect(token)
+    except UnauthorizedError:
+        raise HTTPError(http.UNAUTHORIZED)
 
 
-def connect_from_settings_or_403(user_settings):
-    return connect_or_403(
-        user_settings.dataverse_username,
-        user_settings.dataverse_password
-    ) if user_settings else None
+def connect_from_settings_or_401(user_settings):
+    return connect_or_401(user_settings.api_token) if user_settings else None
 
 
 def delete_file(file):
-    study = file.study
-    study.delete_file(file)
+    dataset = file.dataset
+    dataset.delete_file(file)
 
 
-def upload_file(study, filename, content):
-    study.upload_file(filename, content)
+def upload_file(dataset, filename, content):
+    dataset.upload_file(filename, content)
 
 
-def get_file(study, filename, released=False):
-    return study.get_file(filename, released)
+def get_file(dataset, filename, published=False):
+    return dataset.get_file(filename, published)
 
 
-def get_file_by_id(study, file_id, released=False):
-    return study.get_file_by_id(file_id, released)
+def get_file_by_id(dataset, file_id, published=False):
+    return dataset.get_file_by_id(file_id, published)
 
 
-def get_files(study, released=False):
-    return study.get_files(released)
+def get_files(dataset, published=False):
+    return dataset.get_files(published)
 
 
-def release_study(study):
-    return study.release()
+def publish_dataverse(dataverse):
+    try:
+        dataverse.publish()
+    except OperationFailedError:
+        raise HTTPError(http.BAD_REQUEST)
+
+
+def publish_dataset(dataset):
+    if dataset.get_state() == 'RELEASED':
+        raise HTTPError(http.CONFLICT)
```

Comment:
Error messages here are communicated to the user in javascript. For example, see https://github.com/rliebz/osf.io/blob/c7398403d0350919e43fa075e4b06bfc365b2808/website/addons/dataverse/static/dataverseFangornConfig.js#L62-L71.

Either way, I've added basic error messages to the client as well.